### ISSUE TITLE
Windows build

### ISSUE THREAD

**Issue by Groutcho** [connormanning/entwine#38]
I need to build Entwine to run on Windows desktop, in order to build Cesium 3D Tiles for moderately sized point clouds (around 50-500 MB).

I encountered a lot of errors (2000+) during the build using MSVC. I could fix most of them, although I'm by no mean a C++ expert, so this will probably need a bit of code review. They mainly revolve around the known conflict around the `std::min()`/`std::max()` clashing with the macros defined in `windows.h`, but also some less obvious differences in accessing private members.

* http://stackoverflow.com/q/13416418

The CMake configuration will also probably need some updating to add some paths to dependencies (lzma...)

Maybe also add an AppVeyor CI config.

**Comment by Groutcho** [comment]
I finally could build and link entwine, but not without some hacking:

- I had to disable lzma compression (comment out `lzma.cpp` and `[de]compressLzma()`) due to some linker issues (missing symbols). I used `xz` to get `lzma.h` and `lzma.lib`, but with no success.
- I had to manually link the various libraries in `entwine/` (`formats`, `reader`, `third`...) by adding their absolute path in the `.vcxproj` files, because they werent linked at all. It looks like, in the root `CMakeLists.txt`, we have a `add_library(entwine SHARED ${OBJS})` that doesn't seem to work on Windows.
- Sprinkle some `#include <algorithm>`, `#include <string>`, `#ifndef _WIN32` (especially for`stack-trace.hpp`)
- I had to tell CMake how to locate some dependencies (`pdalcpp.lib`...)

Once the build succeeded, there were still some issues with Arbiter (notably `mkdirp` not yet implemented)

@hobu , I could compile PDAL using your Appveyor `config.cmd` file, so I used this as a basis for Entwine.

Can I push a WIP pull request to review my findings ?

It would be awesome if upstream Entwine was able to build for Windows 😄 

Thank you very much,

Sébastien
